tests/50federation/00prepare.pl

```diff
@@ -169,6 +169,135 @@ sub send_and_await_event {
 push @EXPORT, qw( send_and_await_event );
 
 
+=head2 send_and_await_outlier
+
+   send_and_await_outlier(
+      $inbound_server, $outbound_client, $room, $sending_user_id, $receiving_user,
+   )->then( sub {
+      my ( $outlier_event ) = @_;
+   });
+
+Arranges for an outlier event to be sent over federation.
+
+I<$inbound_server> should be a L<SyTest::Federation:Server>, likely
+I<$main::INBOUND_SERVER>, which will handle the incoming federation requests
+involved.
+
+I<$outbound_client> should be a L<SyTest::Federation::Client>, most likely
+I<$main::OUTBOUND_CLIENT>, which is used to send the event.
+
+I<$room> should be the L<SyTest::Federation::Room> in which to send the event.
+
+I<$sending_user_id> is a user on the Sytest federation server, and should be a
+member of the room.
+
+I<$receiving_user> should be a L<User> on the server under test, which will
+poll for receiving the event. Must also be a member of the room.
+
+The created outlier is returned.
+
+=cut
+
+sub send_and_await_outlier {
+   my ( $inbound_server, $outbound_client, $room, $sending_user_id, $receiving_user ) = @_;
+
+   # to construct an outlier, we create three events, Q, R, S.
+   #
+   # We send S over federation, and allow the server to backfill R, leaving
+   # the server with a gap in the dag. It therefore requests the state at Q,
+   # which leads to Q being persisted as an outlier.
+
+   my $first_home_server = $receiving_user->server_name;
+   my $room_id = $room->room_id;
+   my %initial_room_state  = %{ $room->{current_state} };
+
+   my ( $outlier_event_Q, $outlier_event_id_Q ) = $room->create_and_insert_event(
+      type => 'm.room.member',
+      sender => $sending_user_id,
+      state_key => $sending_user_id,
+      content => { membership => 'join' },
+      auth_events => $room->make_event_refs(
+         $room->get_current_state_event( "m.room.create" ),
+         $room->get_current_state_event( "m.room.power_levels" ),
+         $room->get_current_state_event( "m.room.member", $sending_user_id ),
```

Comment:
Looking at `Room->create_event`, I think these are essentially the default `auth_events` (although that seems to also specify `m.room.join_rules`). Am I missing something or are we just being specific here?

Reply:
yeah, I don't know why I did this.